src/control/server/ctl_ranks_rpc_test.go

```diff
@@ -667,22 +631,14 @@ func TestServer_CtlSvc_ResetFormatRanks(t *testing.T) {
 			// no results as rank can't be read from superblock
 			expResults: []*sharedpb.RankResult{},
 		},
-		"missing ranks": {
-			req:        &ctlpb.RanksReq{Ranks: "0,3"},
-			expResults: []*sharedpb.RankResult{},
-		},
-		"context timeout": { // near-immediate parent context Timeout
-			req:        &ctlpb.RanksReq{Ranks: "0-3"},
-			ctxTimeout: time.Nanosecond,
-			expErr:     context.DeadlineExceeded, // parent ctx timeout
-		},
 		"instances already started": {
 			req:              &ctlpb.RanksReq{Ranks: "0-3"},
 			instancesStarted: true,
 			expErr:           FaultInstancesNotStopped("reset format", 1),
 		},
 		"instances reach wait format": {
-			req: &ctlpb.RanksReq{Ranks: "0-3"},
+			req:     &ctlpb.RanksReq{Ranks: "0-3"},
+			timeout: 5 * time.Second,
```

Comment:
Shouldn't need a timeout for this case, should we?

Reply:
yes we do neeed a timeout for this case because we need to wait for the requestStart() function to be called and then unblocked by the test go-routine reading from the startRequested channel, in a constrained environment this might take a non-negligible amount of time